src/runtime_src/core/tools/common/ReportAieShim.cpp

```diff
@@ -0,0 +1,383 @@
+                    "PC_0",
+                    "Memory_Stall"
+                ],
+                "Memory_module": [
+                    "Lock_0_Acquired",
+                    "DMA_S2MM_0_go_to_idle"
+                ],
+                "PL_module": [
+                    "DMA_S2MM_0_Error",
+                    "Lock_0_Acquired"
+                ]
+            }
+        },
+        ....
+*/
+void
+populate_aie_shim(const xrt_core::device * device, const std::string& desc, boost::property_tree::ptree& pt)
+{
+  pt.put("description", desc);
+  boost::property_tree::ptree _pt;
+
+  try {
+  std::string aie_data = xrt_core::device_query<qr::aie_shim_info>(device);
+    std::stringstream ss(aie_data);
+    boost::property_tree::read_json(ss, _pt);
+  } catch (const std::exception& ex){
+    pt.put("error_msg", ex.what());
+    return;
+  }
+
+  try {
+    boost::property_tree::ptree tile_array;
+    for (auto& as: _pt.get_child("aie_shim")) {
+      boost::property_tree::ptree& oshim = as.second;
+      boost::property_tree::ptree ishim;
+      int col = oshim.get<uint32_t>("col");
+      int row = oshim.get<uint32_t>("row");
+      
+      ishim.put("column", col);
+      ishim.put("row", row);
+      int count = 0;
+
+      boost::property_tree::ptree empty_pt;
+      for (auto& node: oshim.get_child("dma.Channel_status.MM2S", empty_pt)) {
+        ishim.put("dma.channel_status.mm2s.channel_"+std::to_string(count++), node.second.data());
+      }
+
+      count = 0;
+      for (auto& node: oshim.get_child("dma.Channel_status.S2MM", empty_pt)) {
+        ishim.put("dma.channel_status.s2mm.channel_"+std::to_string(count++), node.second.data());
+      }
+
+      count = 0;
+      std::string mode;
+      for (auto& node: oshim.get_child("dma.Mode.MM2S", empty_pt)) {
+        mode +=(mode.empty()?"":", ")+node.second.data();
+      }
+
+      if(!mode.empty())
+        ishim.put("dma.mode.mm2s", mode);
+
+      mode.clear();
+      for (auto& node: oshim.get_child("dma.Mode.S2MM", empty_pt)) {
+        mode +=(mode.empty()?"":", ")+node.second.data();
+      }
+
+      if(!mode.empty())
+        ishim.put("dma.mode.s2mm", mode);
+
+      addnode("dma.Queue_size.MM2S", "dma.queue_size.mm2s.channel_", oshim, ishim);
+      addnode("dma.Queue_size.S2MM", "dma.queue_size.s2mm.channel_", oshim, ishim);
+      addnode("dma.Queue_status.MM2S", "dma.queue_status.mm2s.channel_", oshim, ishim);
+      addnode("dma.Queue_status.S2MM", "dma.queue_status.s2mm.channel_", oshim, ishim);
+      addnode("dma.Current_BD.MM2S", "dma.current_bd.mm2s.channel_", oshim, ishim);
+      addnode("dma.Current_BD.S2MM", "dma.current_bd.s2mm.channel_", oshim, ishim);
+      addnode("dma.Lock_ID.MM2S", "dma.lock_id.mm2s.channel_", oshim, ishim);
+      addnode("dma.Lock_ID.S2MM", "dma.lock_id.s2mm.channel_", oshim, ishim);
+
+      addnodelist("lock", "lock.", oshim, ishim);
+      addnodelist("errors.PL_module", "errors.pl_module.", oshim, ishim);

```

Comment:
The error schema concerns me for it isn't predictable.  If there are going to be errors, it should be an array.   Something like:
```
"errors": [
  {
    "name" : "core_module",
    "value" : "AXI-MM_slave_error"
  }
]
```

Reply:
fixed